tests/test_backend_datamart_serving_readiness.py

```
@@ -0,0 +1,482 @@
+
+def _fetch_row_counts(session, service_id) -> dict[str, int]:
+    counts = {}
+    for table_name in EXPECTED_TABLES:
+        counts[table_name] = session.execute(
+            text(
+                f"""
+                SELECT COUNT(*)
+                FROM {table_name}
+                WHERE date = :target_date
+                  AND service_id = :service_id
+                """
```

Ruff (0.15.12)

[error] Possible SQL injection vector through string-based query construction

(S608)